src/server/lib/validateFromUri.ts

```diff
@@ -0,0 +1,12 @@
+import { logger } from './serverSideLogger';
+
+export const validateFromUri = (
+	fromURI: string | undefined,
+): string | undefined => {
+	if (!fromURI?.startsWith('/oauth2/')) {
+		logger.warn('Got invalid fromURI from request.', undefined, { fromURI });
```

Comment:
Just so we can keep an eye on this after merging to make sure nothing breaks, maybe its a bit overcautious?